### ISSUE TITLE
[operator] Provide checluster template with default values

### ISSUE THREAD

**Issue by benoitf** [eclipse-che/che#13930]
### Is your enhancement related to a problem? Please describe.
If you go on https://operatorhub.io/operator/eclipse-che and click on `Yaml Example` there is a small subset of what can be changed but for example you may have no idea about what the default images are
For example there is

```identityProviderImage: ''```
but you don't know what will be used so it will be nice if it's in the default value or in a comment just before this line

### Describe the solution you'd like
Add values as comments of the yaml example or as default values

### Describe alternatives you've considered
none

### Additional context
It might be the same case for che-operator for openshift

**Comment by l0rd** [comment]
I think we can close this issue because it's fixed by https://github.com/eclipse/che-operator/pull/53

**Comment by l0rd** [comment]
cc @davidfestal

**Comment by davidfestal** [comment]
Well, it's not really fixed by eclipse/che-operator#53. This PR only did this for the nightly channel (of the preview packages), and additionaly, comments will not be kept in these examples that show up in the OperatorHub, since thy are defined as embeded json in the OLM CSV files.

What we should probably do anyway is:
- Generate a CRD (the definition) with an OpenApi schema, so that, at least for Openshift, the syntax will be proposed when adding a custom resource. The comments in the Go code should be the documentation of each field in the CRD OpenApi schema.
- But that still wouldn't probably help in the Kubernetes case where the examples ate not editable

To summarize this is not that easy and we should think more in-depth about what we really want here.
